catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/WhenNfcBrickTest.java

```diff
@@ -0,0 +1,488 @@
+		Assert.assertTrue("Timeout while waiting for user variable change - 'tag id'",
+				userVariableEqualsWithinTimeout(readTagId, Double.parseDouble(TAG_ID_2), 2000));
+		Assert.assertTrue("Timeout while waiting for user variable change - 'tag message' ",
+				userVariableEqualsWithinTimeout(readTagMessage, NFC_NDEF_STRING_2, 2000));
+	}
+
+	@Test
+	public void testSelectTagAndPlay() {
+		checkIfSpinnerOnBrickAtPositionShowsString(R.id.brick_when_nfc_spinner, nfcBrickPosition,
+				R.string.brick_when_nfc_default_all);
+		clickSpinnerValueOnBrick(R.id.brick_when_nfc_spinner, nfcBrickPosition, R.string.test_tag_name_1);
+
+		onView(withId(R.id.button_play))
+				.perform(click());
+
+		Script script = ProjectManager.getInstance().getCurrentScene().getSpriteList().get(0).getScript(0);
+		assertEquals("tag not set", ((WhenNfcBrick) script.getScriptBrick()).getNfcTag().getNfcTagName(),
+				tagDataList.get(0).getNfcTagName());
+		pressBack();
+		pressBack();
+		checkIfSpinnerOnBrickAtPositionShowsString(R.id.brick_when_nfc_spinner, nfcBrickPosition,
+				R.string.test_tag_name_1);
+		clickSpinnerValueOnBrick(R.id.brick_when_nfc_spinner, nfcBrickPosition, R.string.test_tag_name_2);
+
+		onView(withId(R.id.button_play))
+				.perform(click());
+
+		script = ProjectManager.getInstance().getCurrentScene().getSpriteList().get(0).getScript(0);
+		assertEquals("tag not set", ((WhenNfcBrick) script.getScriptBrick()).getNfcTag().getNfcTagName(),
+				tagDataList.get(1).getNfcTagName());
+		pressBack();
+		pressBack();
+		checkIfSpinnerOnBrickAtPositionShowsString(R.id.brick_when_nfc_spinner, nfcBrickPosition, R.string
+				.test_tag_name_2);
+	}
+
+	@Test
+	public void testSpinnerUpdatesDelete() {
+		List<String> spinnerValuesResourceIdsContained = Arrays.asList(
+				getResourcesString(R.string.brick_when_nfc_default_all),
+				getResourcesString(R.string.new_nfc_tag),
+				getResourcesString(R.string.test_tag_name_1),
+				getResourcesString(R.string.test_tag_name_2));
+		List<String> spinnerValuesResourceIdsNotContained = Arrays.asList();
+
+		checkNfcSpinnerContains(nfcBrickPosition, spinnerValuesResourceIdsContained, spinnerValuesResourceIdsNotContained);
+
+		UiHardwareTestUtils.gotoNfcFragment(nfcBrickPosition);
+
+		contextMenuActionDelete(getResourcesString(R.string.test_tag_name_1));
+
+		pressBack();
+
+		spinnerValuesResourceIdsContained = Arrays.asList(
+				getResourcesString(R.string.brick_when_nfc_default_all),
+				getResourcesString(R.string.new_nfc_tag),
+				getResourcesString(R.string.test_tag_name_2));
+		spinnerValuesResourceIdsNotContained = Arrays.asList(getResourcesString(R.string.test_tag_name_1));
+
+		checkNfcSpinnerContains(nfcBrickPosition, spinnerValuesResourceIdsContained,
+				spinnerValuesResourceIdsNotContained);
+	}
+
+	@Test
+	public void testSpinnerUpdatesRename() {
+		String renamedTag = "tag_renamed";
+		List<String> spinnerValuesContained = Arrays.asList(
```

Comment:
I'm afraid that this is not possible since newly read NFC tags do not get an ID that would be inside the resource IDs (e.g. "nfc_tag" is but "nfc_tag_1", "nfc_tag_2", ... are not.)